vita3k/modules/SceNet/SceNet.cpp

```diff
@@ -258,14 +258,37 @@ EXPORT(Ptr<int>, sceNetErrnoLoc) {
     return addr.cast<int>();
 }
 
-EXPORT(int, sceNetEtherNtostr) {
-    TRACY_FUNC(sceNetEtherNtostr);
-    return UNIMPLEMENTED();
+EXPORT(int, sceNetEtherNtostr, SceNetEtherAddr *n, char *str, uint len) {
+    TRACY_FUNC(sceNetEtherNtostr, n, str, len);
+    if (!emuenv.net.inited)
+        return RET_ERROR(SCE_NET_ERROR_ENOTINIT);
+
+    if (!n || !str)
+        return RET_ERROR(SCE_NET_ERROR_EINVAL);
+
+    if (0x11 < len) {
+        snprintf(str, len, "%02x:%02x:%02x:%02x:%02x:%02x",
+            n->data[0], n->data[1], n->data[2], n->data[3], n->data[4], n->data[5]);
+        return 0;
+    }
+    return RET_ERROR(SCE_NET_ERROR_EINVAL);
```

Comment:
code flow looks something strange for me
```suggestion
    if (len <=0x11) 
        return RET_ERROR(SCE_NET_ERROR_EINVAL);
        
    snprintf(str, len, "%02x:%02x:%02x:%02x:%02x:%02x",
        n->data[0], n->data[1], n->data[2], n->data[3], n->data[4], n->data[5]);
    return 0;
```

Reply:
should be better now